packages/core/src/components/tag-input/tagInput.tsx

```diff
@@ -222,15 +226,15 @@ export class TagInput extends AbstractPureComponent<ITagInputProps, ITagInputSta
                 [Classes.DISABLED]: disabled,
                 [Classes.FILL]: fill,
                 [Classes.LARGE]: large,
+                // add CSS class when empty so the placeholder padding can match that of an empty <InputGroup>
+                // (also, no need for this to be part of the public Classes API)
+                // see: https://github.com/palantir/blueprint/issues/2872
+                [`${Classes.TAG_INPUT}-empty`]: !isSomeValueDefined,
```

Comment:
what about if it has a `leftIcon`?

this can also be accomplished using `:first-child` twice - on input-values and input.

Reply:
![image](https://user-images.githubusercontent.com/464822/45171566-97e07000-b1b7-11e8-899e-52f420ccaddb.png)
 
too much left padding